### ISSUE TITLE
Autoscale LR, --gpus, --gpu_ids, --gpu_id confusion

### ISSUE THREAD

**Issue by Divadi** [open-mmlab/mmdetection3d#1308]
Based on these lines:
https://github.com/open-mmlab/mmdetection3d/blob/fd3112bc4954071456660dbb3904a6c9b7a54d86/tools/train.py#L42-L59
and their usage here:
https://github.com/open-mmlab/mmdetection3d/blob/fd3112bc4954071456660dbb3904a6c9b7a54d86/tools/train.py#L131-L147

I am a bit confused on what to use, and how it relates to autoscale-lr.

In previous versions, I would use --gpus 3 --autoscale-lr to automatically scale the default LR (intended for 8 gpus) to my 3 gpu setting.

Now:
1. The tooltip for --gpus and --gpu-ids indicates "--gpu-id" should be used instead. However, I'm not sure what --gpu-id is; the tooltip for it says "number of gpus to use, (only applicable to non-distributed training)". How can multiple GPUs be used for non-distributed training, and is --gpu-id supposed to indicate the ID(s?) of the gpus used or the number of gpus?
2. Following the tooltip for --gpu-id and setting it to 3 (I'm using 3 gpus), here:
https://github.com/open-mmlab/mmdetection3d/blob/fd3112bc4954071456660dbb3904a6c9b7a54d86/tools/train.py#L142-L147
cfg.gpu_ids is now a singleton list, containing the # of gpus to use. However, this singleton list's length of 1 is used to scale the learning rate.
3. Assuming that the tooltip for gpu-id is wrong and gpu-id is supposed to specify a single gpu that's used, the above makes sense I think. Then, however, there doesn't seem to be a way to use autoscale-lr with multiple gpus. If I set --gpus, then cfg.gpu_ids gets set to range(1). If I use cfg.gpu-ids with multiple ids, it gets overwritten as well.
https://github.com/open-mmlab/mmdetection3d/blob/fd3112bc4954071456660dbb3904a6c9b7a54d86/tools/train.py#L131-L141

**Comment by ZwwWayne** [comment]
MMDet already deprecated auto-scale-lr long time ago. It is planing for adding the feature back, with more robust and reasonable design. See https://github.com/open-mmlab/mmdetection/pull/7482, this PR will be merged in this release.

**Comment by Divadi** [comment]
I see, based on batch size. Thank you for letting me know
